pkg/operatorsource/updater.go

```diff
@@ -0,0 +1,113 @@
+package operatorsource
+
+import (
+	"context"
+	"errors"
+
+	"github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+	"github.com/operator-framework/operator-marketplace/pkg/appregistry"
+	"github.com/operator-framework/operator-marketplace/pkg/datastore"
+	"github.com/operator-framework/operator-marketplace/pkg/phase"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Updater is an interface that can be used to check whether a remote registry
+// has any update(s) and trigger a rebuild of in cluster OperatorSource cache.
+type Updater interface {
```

Comment:
Given we talk about CR update reconciliation, can we come up with another name for this interface and struct to signify that you are updating the underlying datastore and not the CR itself?

Reply:
`RegistryUpdater` ok?

Reply:
`RegistryUpdater` seems to imply that you are updating the registry. How about `CacheUpdater`?